### PR TITLE
stream: add error logging on transformation step

### PR DESCRIPTION
**Pull request by zzacharo**
:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_rdm_migrator/streams/runner.py [hunk 1]
```diff
@@ -63,9 +71,11 @@ def run(self):
...
                 stream.run()
-            finally:
+                # sucessfully finished stream run, now we can dump that stream cache
                 for name, cache in self.cache.items():
                     if name == "communities":  # FIXME: implement communities cache
                         continue
                     cache_file = self.cache_dir / f"{name}.json"
                     cache.dump(cache_file)
+            except Exception as exc:
+                print(exc)
```
Comment (ppanero) [inline invenio_rdm_migrator/streams/runner.py:81]: If there is no continuation, this has the same effect that no try/except block. No? (i.e. exception printed in the terminal)

File: invenio_rdm_migrator/streams/runner.py [hunk 2]
```diff
@@ -31,6 +32,13 @@ def __init__(self, stream_definitions, config_filepath):
         self.tmp_dir.mkdir(parents=True, exist_ok=True)
         self.cache_dir = Path(config.get("cache_dir"))
         self.cache_dir.mkdir(parents=True, exist_ok=True)
+        self.errorlog_dir = Path(config.get("errorlog_dir"))
+        self.errorlog_dir.mkdir(parents=True, exist_ok=True)
+        logger = logging.getLogger("migration")
+        logger.setLevel(logging.ERROR)
```
Comment (ppanero) [inline invenio_rdm_migrator/streams/runner.py:38]: curiosity: do we need to set the level in the logger AND the file handler? wouldn't it be enough with one of them?

File: invenio_rdm_migrator/streams/cache.py [hunk 3]
```diff
@@ -116,7 +116,8 @@ def update(self, key, data):
 
         nd_id = data.get("next_draft_id")  # not present in records
         if nd_id:
-            assert not parent["next_draft_id"]  # it can only happen once
+            # TODO: PUT ME BACK!!!
```
Comment (ppanero) [inline invenio_rdm_migrator/streams/cache.py:119]: 😂

File: invenio_rdm_migrator/streams/runner.py [hunk 4]
```diff
@@ -31,6 +32,13 @@ def __init__(self, stream_definitions, config_filepath):
         self.tmp_dir.mkdir(parents=True, exist_ok=True)
         self.cache_dir = Path(config.get("cache_dir"))
         self.cache_dir.mkdir(parents=True, exist_ok=True)
+        self.errorlog_dir = Path(config.get("errorlog_dir"))
```
Comment (ppanero) [inline invenio_rdm_migrator/streams/runner.py:35]: I'd call it `log_dir`, it should be independent of the level 😅 #picky

**Review by ppanero — APPROVED** [review]

File: invenio_rdm_migrator/streams/cache.py (hunk 3 above)
Comment (zzacharo) [reply invenio_rdm_migrator/streams/cache.py:119]: oups

File: invenio_rdm_migrator/streams/runner.py (hunk 2 above)
Comment (zzacharo) [reply invenio_rdm_migrator/streams/runner.py:38]: yes as there are 2 levels of configuration, one for the handler and one for the logger...

**Pull request merged**